library/data/recipe/src/main/java/org/quiltmc/qsl/recipe/api/serializer/QuiltRecipeSerializer.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.recipe.api.serializer;
+
+import com.google.gson.JsonObject;
+
+import net.minecraft.recipe.Recipe;
+import net.minecraft.recipe.RecipeSerializer;
+
+/**
+ * Represents a recipe serializer for mods to implement.
+ * <p>
+ * This will allow serialization to JSON of recipes. Useful for recipe dumping.
+ *
+ * @param <T> the recipe
+ * @version 1.0.0
+ * @since 1.0.0
+ */
+public interface QuiltRecipeSerializer<T extends Recipe<?>> extends RecipeSerializer<T> {
```

Comment:
To ensure all the serializers have this, should 1. interface injection be used (once the generic issue is fixed) and/or 2. a runtime check to warn that the API isn't implemented?

Reply:
Generic issue doesn't seem to be a high priority in loom, so might do the runtime check instead.

Reply:
A warning log has been added in dev environment or with the `quilt.recipe.debug` property.